pkg/mutation/assignmeta_mutator.go

```diff
@@ -42,16 +43,15 @@ type AssignMetadataMutator struct {
 }
 
 // assignMetadataMutator implements mutator
-var _ Mutator = &AssignMetadataMutator{}
+var _ MutatorWithSchema = &AssignMetadataMutator{}
```

Comment:
I think it either makes sense to have assignmeta belong to mutator, and have mutator implement the `Path` method, or get rid of Mutator since MutatorWithSchema will be the only interface implemented by objects.

Reply:
We definitely can't have metadata mutators use `MutatorWithSchema`.

The `Path()` method should probably have gone into the `Mutator` interface. I think the defining method for `MutatorWithSchema()` is actually `SchemaBindings()`

Reply:
Done